src/tiled/scriptdialog.cpp

```diff
@@ -0,0 +1,330 @@
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "scriptdialog.h"
+#include "scriptmanager.h"
+#include <QProcessEnvironment>
+#include <QJSEngine>
+#include <string.h>
+#include <typeinfo>
+#include <memory>
+
+namespace Tiled {
+
+ScriptDialog::ScriptDialog(): ScriptDialog(QString(), 450, 450)
+{
+    
+}
+ScriptDialog::ScriptDialog(const QString &title, const int width=450, const int height=450)
+ : QDialog(MainWindow::maybeInstance())
+{
+    if (title.isEmpty()){
+        setTitle(QString::fromUtf8("Script"));
+    } else {
+      setTitle(title);
+    }
+    setAttribute(Qt::WA_DeleteOnClose);
+    m_widgetNumber = 0;
+    m_lastWidgetTypeName.clear();
+
+    setMinimumSize(width, height);
+    setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
+    initializeLayout();
+    QMetaObject::connectSlotsByName(this);
+}
+void ScriptDialog::initializeLayout()
+{
+    m_gridLayoutWidget = new QWidget(this);
+    m_gridLayoutWidget->setObjectName(QString::fromUtf8("gridLayoutWidget"));
+    m_gridLayoutWidget->setGeometry(QRect(15, 15, minimumWidth()-30, minimumHeight()-30));
+    m_gridLayout = new QGridLayout(m_gridLayoutWidget);
+    m_gridLayout->setContentsMargins(15, 15, 15, 15);
+    m_gridLayout->setObjectName(QString::fromUtf8("m_gridLayout"));
+    // make the right-hand column more likely to stretch
+    m_gridLayout->setColumnStretch(0, m_leftColumnStretch);
+    m_gridLayout->setColumnStretch(1, m_rightColumnStretch);
+    m_rowIndex = 0;
+    addNewRow();
+}
+void ScriptDialog::clear()
+{
+    m_widgetNumber = 0;
+    QLayoutItem * item;
+    QLayout * sublayout;
+    QWidget * widget;
+    while ((item = m_gridLayout->takeAt(0))) {
+        if ((sublayout = item->layout()) != 0) {/* do the same for sublayout*/}
+        else if ((widget = item->widget()) != 0) {widget->hide(); delete widget;}
+        else {delete item;}
+    }
+    delete m_gridLayoutWidget;
+    initializeLayout();
+}
+void ScriptDialog::resize(const int width, const int height)
+{
+    QSize oldSize = size();
+    QDialog::resize(width, height);
+    QSize newSize = QSize(width, height);
+    m_gridLayout->setGeometry(QRect(15,15, width-30, height-30));
+    m_gridLayoutWidget->setGeometry(QRect(15,15, width-30, height-30));
+    setMinimumWidth(width);
+    setMinimumHeight(height);
+    adjustSize();
+    QResizeEvent event = QResizeEvent(newSize, oldSize);
+    resizeEvent(&event);
```

Comment:
Hmm, this whole function should just be calling `Dialog::resize(width, height)` and be done with it. The way to make the internal grid layout widget resize along with the dialog is by putting it in a layout and setting that layout on the dialog.

Likely, it would be enough to set the `m_gridLayout` as the layout of the dialog, in which case we don't need `m_gridLayoutWidget` at all.

Reply:
Removed resize()